src/simplewallet/simplewallet.cpp

```diff
@@ -8937,21 +8937,31 @@ bool simple_wallet::verify(const std::vector<std::string> &args)
   return true;
 }
 //----------------------------------------------------------------------------------------------------
-bool simple_wallet::export_key_images(const std::vector<std::string> &args)
+bool simple_wallet::export_key_images(const std::vector<std::string> &args_)
 {
   if (m_wallet->key_on_device())
   {
     fail_msg_writer() << tr("command not supported by HW wallet");
     return true;
   }
-  if (args.size() != 1)
+  auto args = args_;
+
+  if (m_wallet->watch_only())
   {
-    PRINT_USAGE(USAGE_EXPORT_KEY_IMAGES);
+    fail_msg_writer() << tr("wallet is watch-only and cannot export key images");
     return true;
   }
-  if (m_wallet->watch_only())
+
+  bool all = false;
+  if (args.size() >= 2 && args[0] == "all")
   {
-    fail_msg_writer() << tr("wallet is watch-only and cannot export key images");
+    all = true;
+    args.erase(args.begin());
+  }
+
+  if (args.size() != 1)
```

Comment:
```suggestion
  if (args.size() != 1 && args[0] != "all")
```
But leaving it like it is and accepting all as a valid file name would work too.

Reply:
That seems wrong. It'd accept, eg: export_key_image all bar baz I think.

Reply:
Sorry I meant to type `if (args.size() != 1 || args[0] != "all")` but typing `export_key_image all` is theoretically valid so I’ll just approve the PR.